implementations/rust/ockam/ockam/src/profile.rs

```diff
@@ -191,18 +191,98 @@ impl Profile {
     pub const CURRENT_CHANGE_VERSION: u8 = 1;
 }
 
-impl Profile {
+pub trait ProfileIdentity {
     /// Return unique [`Profile`] identifier, which is equal to sha256 of the root public key
-    pub fn identifier(&self) -> &ProfileIdentifier {
+    fn identifier(&self) -> &ProfileIdentifier;
+}
+
+pub trait ProfileChanges {
+    /// Return change history chain
+    fn change_events(&self) -> &[ProfileChangeEvent];
+    /// Add a change event.
+    fn update_no_verification(
+        &mut self,
+        change_event: ProfileChangeEvent,
+    ) -> ockam_core::Result<()>;
+
+    /// Verify the whole change event chain
+    fn verify(&mut self) -> ockam_core::Result<()>;
```

Comment:
This could also use a more descriptive name. `verify_changes` or `verify_profile`?